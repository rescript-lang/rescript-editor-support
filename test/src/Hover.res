let abc = 22 + 34
//  ^hov

type t = (int, float)
//   ^hov

module Id = {
//     ^hov
  type x = int
}

@ocaml.doc("This module is commented")
module Dep: {
  @ocaml.doc("Some doc comment")
  let customDouble: int => int
} = {
  let customDouble = foo => foo * 2
}

module D = Dep
//         ^hov

let cd = D.customDouble
//         ^hov

module HoverInsideModuleWithComponent = {
  let x = 2 // check that hover on x works
//    ^hov
  @react.component
  let make = () => React.null
}

@ocaml.doc("Doc comment for functionWithTypeAnnotation")
let functionWithTypeAnnotation : unit => int = () => 1
//  ^hov

@react.component
let make = (~name) => React.string(name)
//           ^hov

@react.component
let make2 = (~name:string) => React.string(name)
//           ^hov