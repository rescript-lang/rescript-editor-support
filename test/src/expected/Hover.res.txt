Hover src/Hover.res 0:4
{"contents": "```rescript\nint\n```"}

Hover src/Hover.res 3:5
{"contents": "```rescript\ntype t = (int, float)\n```"}

Hover src/Hover.res 6:7
{"contents": "```rescript\nmodule Id = {\n  type x = int\n}\n```"}

Hover src/Hover.res 19:11
{"contents": "\nThis module is commented\n```rescript\nmodule Dep = {\n  let customDouble: int => int\n}\n```"}

Hover src/Hover.res 22:11
{"contents": "```rescript\nint => int\n```\n\nSome doc comment"}

Hover src/Hover.res 26:6
{"contents": "```rescript\nint\n```"}

Hover src/Hover.res 33:4
{"contents": "```rescript\nunit => int\n```\n\nDoc comment for functionWithTypeAnnotation"}

Hover src/Hover.res 37:13
{"contents": "```rescript\nstring\n```"}

Hover src/Hover.res 41:13
{"contents": "```rescript\nstring\n```"}

